central/src/main/java/org/glowroot/central/util/ClusterManager.java

```diff
@@ -116,14 +116,28 @@ public static ClusterManager create(File confDir, Map<String, String> jgroupsPro
 
         private ClusterManagerImpl(File confDir, String jgroupsConfigurationFile,
                 Map<String, String> jgroupsProperties) {
-            GlobalConfiguration configuration = new GlobalConfigurationBuilder()
-                    .transport()
-                    .defaultTransport()
-                    .addProperty("configurationFile",
-                            getConfigurationFilePropertyValue(confDir, jgroupsConfigurationFile))
-                    .globalJmxStatistics()
-                    .enable()
-                    .build();
+            GlobalConfigurationBuilder builder = new GlobalConfigurationBuilder();
+            builder
+                .transport()
+                .defaultTransport()
+                .addProperty("configurationFile",
+                        getConfigurationFilePropertyValue(confDir, jgroupsConfigurationFile))
+                .jmx()
+                .enable()
+                .serialization()
+                    .marshaller(new JavaSerializationMarshaller())
+                        .allowList()
+                            .addRegexps(
+                                    "org\\.glowroot\\.central\\.repo\\.Immutable.*",
+                                    "org\\.glowroot\\.central\\.util\\.ClusterManager\\$.*",
+                                    "org\\.glowroot\\.central\\.DownstreamServiceImpl\\$.*",
+                                    "org\\.glowroot\\.central\\.Immutable.*",
+                                    "org\\.glowroot\\.wire\\.api\\.model\\..*",
+                                    "com\\.google\\.common\\.collect\\.Immutable.*",
+                                    "com\\.google\\.protobuf\\..*"
+                            );
```

Comment:
This list seems to contain all required allowed class patterns for the serializer of infinispan.
Any hint to improve this list will be very much appreciated :)

Reply:
Ok, I change the four first regexp to `org\.glowroot\.central\..*`